From 5c099757c290b1d899ce22e3870a5ea045a24b90 Mon Sep 17 00:00:00 2001
From: Steve Wainstead <wainstead@gmail.com>
Date: Thu, 20 Jan 2011 19:39:01 +0000
Subject: [PATCH 092/223] "main" must return int, claims g++.

---
 server/client_sysv.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/server/client_sysv.c b/server/client_sysv.c
index 250b1c5..c7e9102 100644
--- a/server/client_sysv.c
+++ b/server/client_sysv.c
@@ -72,7 +72,7 @@ write_all(int fd, const char *buffer, int length)
     }
 }
 
-void
+int
 main(int argc, char **argv)
 {
     const char *connect_file = DEFAULT_CONNECT_FILE;
-- 
1.7.9.5

