From b3ef5308fe67ed1875258fe32a80a69a677b3e24 Mon Sep 17 00:00:00 2001
From: Steve Wainstead <wainstead@gmail.com>
Date: Sat, 13 Aug 2011 21:49:09 +0000
Subject: [PATCH 199/223] Latest configure file.

---
 server/configure |   21 +--------------------
 1 file changed, 1 insertion(+), 20 deletions(-)

diff --git a/server/configure b/server/configure
index a0e3411..18c5243 100755
--- a/server/configure
+++ b/server/configure
@@ -4630,26 +4630,7 @@ $as_echo "none needed" >&6; } ;; #(
 $as_echo "$ac_cv_prog_cc_stdc" >&6; } ;;
 esac
 
-
-echo checking for a working const keyword
-cat << EOF > conftest.c
-#ifdef __GNUC__
-#define const __const__
-#endif
-int foo(const char *x) { return 17 + (x - x); }
-int bar() { int x = foo("foo"); return x; }
-EOF
-if $CC -c conftest.c > conftest.out 2>&1; then
-  if test -s conftest.out; then
-    $as_echo "#define const /**/" >>confdefs.h
-
-  fi
-else
-  $as_echo "#define const /**/" >>confdefs.h
-
-fi
-rm -f conftest*
-
+MOO_CONST
 CPP=""
 ac_fn_cxx_check_header_mongrel "$LINENO" "sys/bsdtypes.h" "ac_cv_header_sys_bsdtypes_h" "$ac_includes_default"
 if test "x$ac_cv_header_sys_bsdtypes_h" = xyes; then :
-- 
1.7.9.5

