From ed1e94fa24446329bf1a789060e4708206e5b348 Mon Sep 17 00:00:00 2001
From: Steve Wainstead <wainstead@gmail.com>
Date: Tue, 19 Jun 2012 23:38:19 -0700
Subject: [PATCH 216/223] I'm marking blocks with leading semicolons to note
 what's done

---
 server/log |   44 ++++++++++++++++++++++----------------------
 1 file changed, 22 insertions(+), 22 deletions(-)

diff --git a/server/log b/server/log
index 08ea8d6..32cadb6 100644
--- a/server/log
+++ b/server/log
@@ -2740,28 +2740,28 @@ Date:   Sun Jun 7 19:28:43 2009 +0000
  server/db_file.c |  150 ++++++++++++++++++++++++++++++++++++------------------
  1 file changed, 101 insertions(+), 49 deletions(-)
 
-commit 3a6febd8cb3a492cd0a598919ea3fbb4b3166985
-Author: Steve Wainstead <wainstead@gmail.com>
-Date:   Sun Jun 7 18:36:44 2009 +0000
-
-    two minor fixes
-
- server/code_gen.c |    4 ++--
- server/db_file.c  |    2 +-
- 2 files changed, 3 insertions(+), 3 deletions(-)
-
-commit 4ecb6adc0266d48a34503988b547c2a6ab64cb49
-Author: Steve Wainstead <wainstead@gmail.com>
-Date:   Sun Jun 7 18:28:36 2009 +0000
-
-    convert catch to _catch to get around c++ keyword
-
- server/code_gen.c  |   12 ++++++------
- server/decompile.c |   20 ++++++++++----------
- server/parser.y    |   10 +++++-----
- server/unparse.c   |   16 ++++++++--------
- 4 files changed, 29 insertions(+), 29 deletions(-)
-
+;  commit 3a6febd8cb3a492cd0a598919ea3fbb4b3166985
+;  Author: Steve Wainstead <wainstead@gmail.com>
+;  Date:   Sun Jun 7 18:36:44 2009 +0000
+;  
+;      two minor fixes
+;  
+;   server/code_gen.c |    4 ++--
+;   server/db_file.c  |    2 +-
+;   2 files changed, 3 insertions(+), 3 deletions(-)
+;  
+;  commit 4ecb6adc0266d48a34503988b547c2a6ab64cb49
+;  Author: Steve Wainstead <wainstead@gmail.com>
+;  Date:   Sun Jun 7 18:28:36 2009 +0000
+;  
+;      convert catch to _catch to get around c++ keyword
+;  
+;   server/code_gen.c  |   12 ++++++------
+;   server/decompile.c |   20 ++++++++++----------
+;   server/parser.y    |   10 +++++-----
+;   server/unparse.c   |   16 ++++++++--------
+;   4 files changed, 29 insertions(+), 29 deletions(-)
+;  
 ;  commit 5a0d34a4692a390b4cb212f2443e5e951b64a3e5
 ;  Author: Steve Wainstead <wainstead@gmail.com>
 ;  Date:   Sun Jun 7 18:23:47 2009 +0000
-- 
1.7.9.5

