From 2468cdec940760a10a233b864b6833507ba7afdb Mon Sep 17 00:00:00 2001
From: Steve Wainstead <wainstead@gmail.com>
Date: Mon, 8 Jun 2009 04:20:53 +0000
Subject: [PATCH 016/223] one _new, and a number of casts made.

---
 server/net_multi.c |   20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/server/net_multi.c b/server/net_multi.c
index 9791101..fca11d8 100644
--- a/server/net_multi.c
+++ b/server/net_multi.c
@@ -124,7 +124,7 @@ network_register_fd(int fd, network_fd_callback readable,
 	    break;
     if (i >= max_reg_fds) {	/* No free slots */
 	int new_max = 2 * max_reg_fds;
-	fd_reg *new = (fd_reg *) mymalloc(new_max * sizeof(fd_reg), M_NETWORK);
+	fd_reg *_new = (fd_reg *) mymalloc(new_max * sizeof(fd_reg), M_NETWORK);
 
 	for (i = 0; i < new_max; i++)
 	    if (i < max_reg_fds)
@@ -432,7 +432,7 @@ static int
 enqueue_output(network_handle nh, const char *line, int line_length,
 	       int add_eol, int flush_ok)
 {
-    nhandle *h = nh.ptr;
+    nhandle *h = (nhandle *)nh.ptr;
     int length = line_length + (add_eol ? eol_length : 0);
     char *buffer;
     text_block *block;
@@ -528,7 +528,7 @@ network_make_listener(server_listener sl, Var desc,
 int
 network_listen(network_listener nl)
 {
-    nlistener *l = nl.ptr;
+    nlistener *l = (nlistener *)nl.ptr;
 
     return proto_listen(l->fd);
 }
@@ -549,7 +549,7 @@ network_send_bytes(network_handle nh, const char *buffer, int buflen,
 int
 network_buffered_output_length(network_handle nh)
 {
-    nhandle *h = nh.ptr;
+    nhandle *h = (nhandle *)nh.ptr;
 
     return h->output_length;
 }
@@ -557,7 +557,7 @@ network_buffered_output_length(network_handle nh)
 void
 network_suspend_input(network_handle nh)
 {
-    nhandle *h = nh.ptr;
+    nhandle *h = (nhandle *)nh.ptr;
 
     h->input_suspended = 1;
 }
@@ -565,7 +565,7 @@ network_suspend_input(network_handle nh)
 void
 network_resume_input(network_handle nh)
 {
-    nhandle *h = nh.ptr;
+    nhandle *h = (nhandle *)nh.ptr;
 
     h->input_suspended = 0;
 }
@@ -617,7 +617,7 @@ network_connection_name(network_handle nh)
 void
 network_set_connection_binary(network_handle nh, int do_binary)
 {
-    nhandle *h = nh.ptr;
+    nhandle *h = (nhandle *)nh.ptr;
 
     h->binary = do_binary;
 }
@@ -635,7 +635,7 @@ network_set_connection_binary(network_handle nh, int do_binary)
 void
 network_set_client_echo(network_handle nh, int is_on)
 {
-    nhandle *h = nh.ptr;
+    nhandle *h = (nhandle *)nh.ptr;
 
     /* These values taken from RFC 854 and RFC 857. */
 #define TN_IAC	255		/* Interpret As Command */
@@ -682,13 +682,13 @@ network_open_connection(Var arglist, server_listener sl)
 void
 network_close(network_handle h)
 {
-    close_nhandle(h.ptr);
+    close_nhandle((nhandle *)h.ptr);
 }
 
 void
 network_close_listener(network_listener nl)
 {
-    close_nlistener(nl.ptr);
+    close_nlistener((nlistener *)nl.ptr);
 }
 
 void
-- 
1.7.9.5

